Sources/Markdown/Parser/LazySplitLines.swift

```diff
@@ -28,7 +28,7 @@ struct LazySplitLines: Sequence {
         private var source: URL?
 
         init<S: StringProtocol>(_ input: S, source: URL?) where S.SubSequence == Substring {
-            self.rawLines = input.split(separator: "\n", maxSplits: Int.max, omittingEmptySubsequences: false)
+            self.rawLines = input.components(separatedBy: CharacterSet.newlines).map { Substring($0) }
```

Comment:
minor: I feel like mapping each `String` to a `Substring` goes against the original intent of storing substrings in `private let rawLines: [Substring]`. 

We should either change the implementation to store the newly allocated strings directly or use `.split(omittingEmptySubsequences: false, whereSeparator: \.isNewline)` which returns substrings that reference the original string storage. 

It's also possible to imagine that this implementation wouldn't split at all, but rather scan until the next newline for each call to `next()` but that would be a larger change to its implementation which may not be worth it.